### ISSUE TITLE
Provide way to override "found other export database files" warning

### ISSUE THREAD

**Issue by jasonhollis** [RhetTbull/osxphotos#1546]
osxphotos export /Volumes/Photos --exiftool --update --ramdb --download-missing --touch-file --use-photokit --finder-tag-keywords --directory "{folder_album}" --post-command-error continue
Using last opened Photos library: /Users/biaginoguzzo/Pictures/Photos Library-1-3-2024.photoslibrary
WARNING: found other export database files in this destination directory branch. This likely means you are attempting to export files into a directory 
that is either the parent or a child directory of a previous export. Proceeding may cause your exported files to be overwritten.
You are exporting to /Volumes/Photos, found .osxphotos_export.db files in:
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-12_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-15_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-11_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-09_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-10_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-14_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-04-02_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-04-22_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-04-15_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-04-29_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-13_0100/.osxphotos_export.db
/Volumes/Photos/@Recently-Snapshot/GMT+10_2024-05-06_0100/.osxphotos_export.db
Do you want to continue? [y/N]: n

It's detecting all of the .osxphotos_export.db in all of the snapshots.  I'm glad it detects this and even happier that I do have snapshots, just in case.  Now, how do I force the run or automate around this issue?  This is killing me.  I have my Shortcut Slink and Plist with Launchctl working perfectly.  I just need to put this to sleep.

I have tried everything that looked obvious.

**Comment by jasonhollis** [comment]
Thanks for your help!  LOL I’ve been running Qnap’s for myself and clients for over twenty years and never thought about killing the snapshot visibility.  That should work.  Because I’m running —update the -F command doesn’t work.  I already tried that one.

**Comment by RhetTbull** [comment]
@jasonhollis I'll add a way to override this warning. In the meantime, you can use `yes | osxphotos export ...` to force the "Y" entry to override the warning.

**Comment by jasonhollis** [comment]
Damn!

I knew would have something in there.  I did try /Y. So it’s just a YES at the end of the line?  That’s good to know but oPromessa<https://github.com/oPromessa> helped me out with something I should have been doing anyway.  As I mentioned I’ve been running Qnaps forever and for a lot of clients.  The snapshot setup has saved the asses of many customers.  Most of these customers are lawyers, barristers, and non-it folks.  Having the snapshots visible in the filesystem on the shares is probably stupid on my part anyway.  They are now hidden on about 12 Qnaps.

I probably need to visit the updates you have made and the whole routine I’m using for Bill with Osxphotos.  I should probably also document the end to end routine I’m using with launchctl, plist, and shortcuts to insure the share is mounted, fire off osxphotos –update and capture the output from the shell to a little running logfile.  In shortcuts it would also be pretty easy to text myself if there were any errors.

I appreciate you tracking my plight!  I should probably remind David Sparks that he needs to check out your code.  I know if we could get him to look, he would be fascinated.  He’s an amateur photographer.

Kind regards,

Jason
From: Rhet Turnbull ***@***.***>
Date: Thursday, 16 May 2024 at 23:03
To: RhetTbull/osxphotos ***@***.***>
Cc: Jason Hollis ***@***.***>, Mention ***@***.***>
Subject: Re: [RhetTbull/osxphotos] Snapshots being detected and are scaring Osxphotos (Issue #1546)

@jasonhollis<https://github.com/jasonhollis> I'll add a way to override this warning. In the meantime, you can use yes | osxphotos export ... to force the "Y" entry to override the warning.

**Comment by RhetTbull** [comment]
> So it’s just a YES at the end of the line? 

Yes, the `yes` command simply prints `y` followed by newline to stdout. If you just type it on the command line you'll get an infinite stream of `y` characters. It's sole purpose is things like this where you need to force a prompt to continue.

**Comment by jasonhollis** [comment]
Legend!

Best regards

Jason

You should check out NextDNS<https://nextdns.io/?from=s8kadggp>
________________________________
